core/templates/dev/head/filters.js

```diff
@@ -236,10 +236,27 @@ oppia.filter('parameterizeRuleDescription', [
         } else if (varType === 'Fraction') {
           replacementText = FractionObjectFactory
             .fromDict(inputs[varName]).toString();
+        } else if (varType === 'SetOfUnicodeString') {
+          replacementText = '[';
+          for (var i = 0; i < inputs[varName].length; i++) {
+            if (i !== 0) {
+              replacementText += ', ';
+            }
+            replacementText += inputs[varName][i];
+          }
+          replacementText += ']';
+        } else if (
+          varType === 'Real' || varType === 'NonnegativeInt') {
+          replacementText = inputs[varName] + '';
         } else {
```

Comment:
You still need "else if (varType === '...') {replacementText = inputs[varName];} else {throw Error(...)}`. I understand this works now but we want to also prevent it breaking in the future as new types are added.

Reply:
Ok.. So in the final else case should I do if (typeof(replacementText) !== 'string') throw '...' ? As there are some varTypes that default to type string which are currently being handled in the final else.

Reply:
In that case, list those varTypes explicitly in the penultimate clause.

Reply:
Ok Will do that.

Reply:
Done